### ISSUE TITLE
Emulator Crashes Page on load

### ISSUE THREAD

**Issue by Cattn** [emulatorjs/emulatorjs#416]
Clicking the Start Game button will load the game like normal, but it seems right after bios loads and before game starts, the emulator will freeze (UI and all) and will also freeze the current page. The page shoots up in memory and CPU usage.  Closing the page is a bit laggy, ~~and console fully stops working during that time.~~

Happens on any game type and multiple versions of emulatorjs. This is a pretty new bug for me, but something like it has happened before just not to this extent, it fixed itself last time eventually. 

If you'd like any logs or whatnot let me know what you need

**Comment by Cattn** [comment]
Tested on both Chrome and Edge issue happens on both

**Comment by Nailington** [comment]
As you know, I'm from your branch. When we added out own games a long time ago, if the linking was wrong and it couldn't load, or the file format was incorrect (hard to do, I know), it would hang like that. I would open the Dev Tools and look in the console on the previous page and check there for errors. Make sure the network tab is open before

**Comment by Cattn** [comment]
Not much to see in network tab

Also tested on firefox same thing

**Comment by Cattn** [comment]
I'm gonna close seems as though it's Chrome Remote Desktop interfering/causing the issue

**Comment by ethanaobrien** [comment]
>I'm gonna close seems as though it's Chrome Remote Desktop interfering/causing the issue

This is not the case. See #407. I *think* this is a browser thing

**Comment by Cattn** [comment]
> > I'm gonna close seems as though it's Chrome Remote Desktop interfering/causing the issue
> 
> This is not the case. See #407. I _think_ this is a browser thing

Yep very odd, through testing for some reason, if chrome remote desktop is open in any form for me (even remoting into myself and not using the tab) it fixes it I have yet to restart my laptop to see if that will fix as I'm doing work, but I'll update after I do that

**Comment by Cattn** [comment]
To add on, the second I disconnect from a session or stop a session where a user is viewing, it immediately goes back to not working have not tested it while in the middle of a game yet, although I don't think it would do anything

**Comment by Cattn** [comment]
> To add on, the second I disconnect from a session or stop a session where a user is viewing, it immediately goes back to not working have not tested it while in the middle of a game yet, although I don't think it would do anything

Did nothing as suspected (Tried reloading emulator itself, pause/play) only after pager reload once closed does it break again

**Comment by ethanaobrien** [comment]
Yeah, this seems to be a browser issue. Haven't gotten a chance to open an issue on the chromium bug page yet

**Comment by Nailington** [comment]
> Yeah, this seems to be a browser issue. Haven't gotten a chance to open an issue on the chromium bug page yet

But is it? He said it happens on Firefox too...



> Not much to see in network tab
> 
> Also tested on firefox same thing

**Comment by Nailington** [comment]
It seems to be the emulator itself, or the file itself, but on his end. I have done the same testing with no problems. He even had issues when downloading a ROM from our hosting link, and running it locally through the traditional Upload a ROM function. Seems to be something off on his end. I don't know though.

**Comment by ethanaobrien** [comment]
This issue seems to be really weird and random. I've had this problem occur and had it be fine on a completely different computer. @Cattn What OS does it crash on? @Nailington What OS does it not crash on (for you)

**Comment by Nailington** [comment]
@ethanaobrien 
```
Microsoft Windows
Version 22H2 (0S Build 22621.674)
@ Microsoft Corporation. All rights reserved.
```
This is a Windows 11 Education machine. I have mainly tested it on Chrome: 
`107.0.5304.107 (Official Build) (64-bit) (cohort: Stable)`

Not sure the issue on @Cattn machine.... I know he runs Win11, but not sure the specifics.

**Comment by Cattn** [comment]
> This issue seems to be really weird and random. I've had this problem occur and had it be fine on a completely different computer. @Cattn What OS does it crash on? @Nailington What OS does it not crash on (for you)

Windows 11 22H2, pretty sure same build as him but personal. (Don't have my laptop on me yet, but I'm like 80% sure). This is the only OS I've experienced this on, but haven't really tested much on other ones

(Same build of chrome)

**Comment by ethanaobrien** [comment]
This is really strange that I've only seen this happening on windows. It could be a problem with the browser (across multiple browsers) with emscripten, with the code, I have no idea and I cannot debug this, it gives me 0 information

**Comment by Cattn** [comment]
My OS build is 22621.819 (Slightly dif from his)

**Comment by Cattn** [comment]
Also to provide an update, without restarting my laptop the emulator now works fine again, I've done nothing much to try and fix specifically, and no CRD sessions between now and then to test. I've played games, and used unity/VSC and that's about it.

**Comment by Cattn** [comment]
>He even had issues when downloading a ROM from our hosting link, and running it locally through the traditional Upload a ROM

I think the downloading issues were more of an internet goof, but yes, uploading it did break still. Very annoying issue as seemingly no way to tell if it's the emulator, or the OS, or the browser

**Comment by Nailington** [comment]
For what it is worth, it seemed to have fixed itself randomly. We have no idea what fixed it, or why it was broken in the first place.

**Comment by dimitrikarpov** [comment]
Hello. Recently I played with compiling retroarch cores trying to make a js wrapper over retroarch js Module. And accidentely found similar bug too. Page just starts to freeze after starting an emulator. This bug is reprodusable almost 90% on my ubuntu laptop and almost never (or never) on my win10 laptop.

And the fun thing is if anywhere playing music (in browser or just locally installed music player app) bug not happened. That is why I did not know about this bug.

Maybe, guys, this can help you somehow and this is the same bug.

Looks like bug in the `queueAudio` function https://github.com/libretro/RetroArch/blob/master/emscripten/library_rwebaudio.js#L57

Because in my case emulator starts fine, game showed first screen, audio makes some cracked sound and page get freezed. Also this never happened if just start retroarch without rom (without command line arguments to start with rom). If retroarch menu starts first and you start rom from the retroarch interface bug never happened.

By the way. I set ```audio_latency: 256``` option in retroarch.cfg (in retroarch web player this value is 96)  and it seems have fixed this bug in windows. I messed up my ubuntu install with new pipewire audio system.

Or, maybe this bug is about retroarch frontend assets. We both don't downloading them.

Sorry about my english

**Comment by ethanaobrien** [comment]
Thanks for the report! This makes sense. I'll look into more and debug this when I get a chance to use a machine I can reproduce this on.

**Comment by ripjaw1219** [comment]
when i run it on replit it says "failed to start game"

**Comment by ethanaobrien** [comment]
@ripjaw1219 see #449

**Comment by Cattn** [comment]
Hello, it's been a while but it seems I've caught the bug again

**Comment by Cattn** [comment]
Chrome 109, Windows 11, Happening on my personal site, the demo page, just about anywhere as far as I've seen that the emulator is located

**Comment by Cattn** [comment]
If there's any tests/things you'd like me to run to help identify the issue I'd be down to help out @ethanaobrien

**Comment by ethanaobrien** [comment]
I might have a chance to debug this today.

**Comment by Cattn** [comment]
Alright, perfect

**Comment by Cattn** [comment]
I think I still have this bug on/off atm

**Comment by ethanaobrien** [comment]
We all do, this bug has not been fixed and I have not had time to troubleshoot it. There is no clear cause, although @dimitrikarpov might be onto something. I'll test his solution when I get a chance, but I don't know when this will be

**Comment by Cattn** [comment]
Someone using my site encountered this bug, and they report that Muting the tab fixed it for them, but unmuting instantly makes it crash again. He is certainly onto something

https://github.com/Cattn/gba/discussions/97#discussioncomment-4965120

**Comment by mwmdragon2** [comment]
This issue is caused by Chormium browsers not being able to autoload content that has sound playing, bug or feature this is the issue, it affects other JS emulators as well and is an issue with the browser itself.. disable auto-start and it will work perfectly.

I'd love the ability to replace the "Start game" Screen background ;)

**Comment by Nailington** [comment]
> This issue is caused by Chormium browsers not being able to autoload content that has sound playing, bug or feature this is the issue, it affects other JS emulators as well and is an issue with the browser itself.. disable auto-start and it will work perfectly.
> 
> I'd love the ability to replace the "Start game" Screen background ;)

That may be the case, but our site has EmuJS setup with the "Start Game" button, so while i think this is related, not fully the solution.

PS: Changing the Start Game button itself would be awesome too

**Comment by ethanaobrien** [comment]
>I'd love the ability to replace the "Start game" Screen background ;)

>PS: Changing the Start Game button itself would be awesome too

Yall could open a new issue for these.

And yeah, this issue even happens when I have the start now button, so I don't think its the browser.

**Comment by ethanaobrien** [comment]
@dimitrikarpov 

Wanted to let you know I found the bug in the rwebaudio driver, fixed it, and submitted a pr to upstream retroarch. It has now been merged! This issue should be fixed in your project if you update retroarch 

Pr: https://github.com/libretro/RetroArch/pull/15449

Everyone else, this bug will be fixed in the next version of emulatorjs

**Comment by Cattn** [comment]
This is awesome! Seeing as this issue has finally been fixed, I'm going to close this issue

**Comment by ethanaobrien** [comment]
I'm gonna keep this open till the new update comes out, just in case anyone else has this issue. Thank you!

**Comment by dimitrikarpov** [comment]
@ethanaobrien Wow! Just WOW! Finally this bug has been fixed. And now I can remove my workaround with playing silence sound on super low volume before emulator starts. Thank You!

**Comment by Nailington** [comment]
I am super excited to see this as well! Can't wait to upgrade our version of EmulatorJS :)

**Comment by ethanaobrien** [comment]
Version 4.0 is out!
